File: frontend/src/main/java/com/innertrack/model/Report.java
package com.innertrack.model;

import java.time.LocalDateTime;

public class Report {
    private int id;
    private int reporterId;
    private int reportedId;
    private String reason; // SPAM | HARASSMENT | INAPPROPRIATE | OTHER
    private String details;
    private String status; // PENDING | REVIEWED | DISMISSED
    private String context; // MESSAGING | COMMUNITY
    private LocalDateTime createdAt;
    private LocalDateTime reviewedAt;
    private Integer reviewedBy;

    // Transient — filled by DAO via JOIN
    private String reporterName;
    private String reportedName;

    public Report() {
    }

    public Report(int reporterId, int reportedId, String reason, String details) {
        this.reporterId = reporterId;
        this.reportedId = reportedId;
        this.reason = reason;
        this.details = details;
        this.status = "PENDING";
        this.context = "MESSAGING"; // Default, overridden if specified
    }

    public Report(int reporterId, int reportedId, String reason, String details, String context) {
        this.reporterId = reporterId;
        this.reportedId = reportedId;
        this.reason = reason;
        this.details = details;
        this.status = "PENDING";
        this.context = context;
    }

    // ── Getters / Setters ─────────────────────────────────────

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getReporterId() {
        return reporterId;
    }

    public void setReporterId(int v) {
        this.reporterId = v;
    }

    public int getReportedId() {
        return reportedId;
    }

    public void setReportedId(int v) {
        this.reportedId = v;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String v) {
        this.reason = v;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String v) {
        this.details = v;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String v) {
        this.status = v;
    }

    public String getContext() {
        return context;
    }

    public void setContext(String context) {
        this.context = context;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime v) {
        this.createdAt = v;
    }

    public LocalDateTime getReviewedAt() {
        return reviewedAt;
    }

    public void setReviewedAt(LocalDateTime v) {
        this.reviewedAt = v;
    }

    public Integer getReviewedBy() {
        return reviewedBy;
    }

    public void setReviewedBy(Integer v) {
        this.reviewedBy = v;
    }

    public String getReporterName() {
        return reporterName;
    }

    public void setReporterName(String v) {
        this.reporterName = v;
    }

    public String getReportedName() {
        return reportedName;
    }

    public void setReportedName(String v) {
        this.reportedName = v;
    }

    public String getReasonLabel() {
        return switch (reason == null ? "" : reason) {
            case "SPAM" -> "Spam";
            case "HARASSMENT" -> "Harcèlement";
            case "INAPPROPRIATE" -> "Contenu inapproprié";
            default -> "Autre";
        };
    }
}

File: backend/src/Entity/Report.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'report')]
class Report
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'reporter_id', referencedColumnName: 'id', nullable: false)]
    private User $reporter;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'reported_id', referencedColumnName: 'id', nullable: false)]
    private User $reported;

    #[ORM\Column(length: 50)]
    private string $reason;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $details = null;

    #[ORM\Column(length: 20, options: ['default' => 'MESSAGING'])]
    private string $context = 'MESSAGING';

    #[ORM\Column(length: 20, options: ['default' => 'PENDING'])]
    private string $status = 'PENDING';

    #[ORM\Column(name: 'created_at', type: 'datetime')]
    private \DateTimeInterface $createdAt;

    #[ORM\Column(name: 'reviewed_at', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $reviewedAt = null;

    #[ORM\Column(name: 'reviewed_by', nullable: true)]
    private ?int $reviewedBy = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }
    public function getReporter(): User { return $this->reporter; }
    public function setReporter(User $reporter): self { $this->reporter = $reporter; return $this; }
    public function getReported(): User { return $this->reported; }
    public function setReported(User $reported): self { $this->reported = $reported; return $this; }
    public function getReason(): string { return $this->reason; }
    public function setReason(string $reason): self { $this->reason = $reason; return $this; }
    public function getDetails(): ?string { return $this->details; }
    public function setDetails(?string $details): self { $this->details = $details; return $this; }
    public function getContext(): string { return $this->context; }
    public function setContext(string $context): self { $this->context = $context; return $this; }
    public function getStatus(): string { return $this->status; }
    public function setStatus(string $status): self { $this->status = $status; return $this; }
    public function getCreatedAt(): \DateTimeInterface { return $this->createdAt; }
    public function getReviewedAt(): ?\DateTimeInterface { return $this->reviewedAt; }
    public function setReviewedAt(?\DateTimeInterface $reviewedAt): self { $this->reviewedAt = $reviewedAt; return $this; }
    public function getReviewedBy(): ?int { return $this->reviewedBy; }
    public function setReviewedBy(?int $reviewedBy): self { $this->reviewedBy = $reviewedBy; return $this; }

    public function isPending(): bool { return $this->status === 'PENDING'; }

    public function getReasonLabel(): string
    {
        return match($this->reason) {
            'SPAM' => 'Spam',
            'HARASSMENT' => 'Harcèlement',
            'INAPPROPRIATE' => 'Contenu inapproprié',
            'OTHER' => 'Autre',
            default => $this->reason,
        };
    }
}
